lib/uknetdev/include/uk/netdev_core.h

```diff
@@ -457,6 +461,46 @@ struct uk_netdev_einfo {
 	const char *ipv4_gw_addr;
 };
 
+struct uk_netdev_tx_stats {
+	/** The total number of bytes of data transmitted by the interface */
+	size_t bytes;
+
+	/** The total number of packets of data transmitted by the interface */
+	size_t packets;
+
+	/** The total number of transmit errors detected by the device driver */
+	size_t errors;
+
+	/** The number of FIFO buffer errors */
+	size_t fifo;
+};
+
+struct uk_netdev_rx_stats {
+	/** The total number of bytes of data received by the interface */
+	size_t bytes;
+
+	/** The total number of packets of data received by the interface */
+	size_t packets;
+
+	/** The total number of receive errors detected by the device driver */
+	size_t errors;
+
+	/** The number of FIFO buffer errors */
+	size_t fifo;
+
+	/** The number of packet framing errors */
+	size_t frame;
+};
+
+struct uk_netdev_stats {
+	struct uk_netdev_tx_stats tx_m;
+	struct uk_netdev_rx_stats rx_m;
+};
+
+#ifdef CONFIG_LIBUKNETDEV_STATS
+int uk_netdev_stats_init(struct uk_netdev *dev);
```

Comment:
We probably should not export this API call outside of the library. Instead, we should create a library-local header in the base folder for cross-referencing between library source files. I even did not see `uk_netdev_stats_init()` in `exportsyms.uk`.